vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -151,9 +150,9 @@ def refresh_metrics_metadata_cache(self):
             allowed_counters = []
             for c in counters:
                 metric_name = format_metric_name(c)
-                if metric_name in ALLOWED_METRICS_FOR_MOR[mor_type] and not is_metric_excluded_by_filters(
-                    metric_name, mor_type, self._config.metric_filters
-                ):
+                if is_metric_allowed(
+                    mor_type, metric_name, self._config.collection_type
+                ) and not is_metric_excluded_by_filters(metric_name, mor_type, self._config.metric_filters):
```

Comment:
What do you think about including the metrics filtering in the `is_metric_allowed` method, unless there is a context where one method would be needed without the other.

Reply:
Hm I tend to like them separate because they are doing different validations, but this did make me realize `is_metric_allowed` should probably be moved to the utils file, which I updated in the PR. I also renamed the function to `is_metric_allowed_for_collection_type` to better describe what it's doing. Let me know if you think this is sufficient or we should try to combine the two into one larger `is_metric_allowed` function